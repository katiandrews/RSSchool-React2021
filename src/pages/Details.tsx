import { useEffect, useState } from "react";
import { getBook } from "../api/book-search";
import { IBook } from "../interfaces/IBook";

interface IProps {
  match: {
    params: { id: string };
  };
}

export function Details({ match }: IProps) {
  const [book, setBook] = useState<IBook>();

  useEffect(() => {
    getBook(match.params.id).then((book) => {
      setBook(book);
    });
  }, [book]);

  return <main>{book?.volumeInfo.title}</main>;
}
