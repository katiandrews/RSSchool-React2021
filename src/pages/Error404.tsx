import errorImg from "../assets/notFound.png";

export function Error404() {
  return (
    <main className="error-page">
      <img src={errorImg} alt="page not found" width="800" />
    </main>
  );
}
