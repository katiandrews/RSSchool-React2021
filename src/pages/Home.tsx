import { useEffect, useState } from "react";
import { getBooks } from "../api/book-search";
import { BookCard } from "../components/BookCard/BookCard";
import { Loading } from "../components/Loading/Loading";
import { Pagination } from "../components/Pagination/Pagination";
import { Search } from "../components/Search/Search";
import { IBook } from "../interfaces/IBook";
import { ISearch } from "../interfaces/ISearch";
import { calcStartIndex } from "../shared/calcStartIndex";
import { calculatePages } from "../shared/calculatePages";

export function Home() {
  const [books, setBooks] = useState<IBook[] | null>(null);
  const [searchState, setSearchState] = useState<ISearch>({
    search: "Harry Potter",
    sort: "relevance",
    quantity: 10,
  });
  const [isLoading, setLoading] = useState(true);
  const [totalPages, setPages] = useState(0);
  const [currentPage, setCurrentPage] = useState(1);

  useEffect(() => {
    setLoading(true);
    const startIndex = calcStartIndex(currentPage, searchState.quantity);
    getBooks(
      searchState.search,
      searchState.sort,
      searchState.quantity,
      startIndex
    ).then((bookshelf) => {
      setLoading(false);
      setBooks(bookshelf.items);
      setPages(calculatePages(bookshelf.totalItems, searchState.quantity));
    });
  }, [currentPage, searchState]);

  return (
    <div>
      <Search setSearchState={setSearchState} />
      <main>
        {isLoading && <Loading />}
        {!isLoading &&
          !books &&
          "Books have not been found, try with another query"}
        {!isLoading &&
          books &&
          books.map((book) => <BookCard key={book.id} book={book} />)}
      </main>
      <Pagination
        currentPage={currentPage}
        pages={totalPages}
        setCurrentPage={setCurrentPage}
      />
    </div>
  );
}
