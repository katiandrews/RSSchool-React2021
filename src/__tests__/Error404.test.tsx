import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import { Error404 } from "../pages/Error404";

test("error page renders correctly", () => {
  const { asFragment } = render(<Error404 />);
  expect(asFragment()).toMatchSnapshot();
});
