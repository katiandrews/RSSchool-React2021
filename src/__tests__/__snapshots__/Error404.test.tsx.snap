// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error page renders correctly 1`] = `
<DocumentFragment>
  <main
    class="error-page"
  >
    <img
      alt="page not found"
      src="test-file-stub"
      width="800"
    />
  </main>
</DocumentFragment>
`;
